Libraries/LLM/LLMModel.swift

```diff
@@ -5,7 +5,7 @@ import MLX
 import MLXNN
 
 // Interface for all LLM Models
-public protocol LLMModel: Module {
+public protocol LLMModel: Module, Sendable {
```

Comment:
I don't think this is really Sendable -- the model is full of MLXArrays, which are mutable and very much _not_ sendable.

We might be able to use `@unchecked Sendable` and document that callers have to be very very careful, but that kind of defeats the purpose of `Sendable`.  I think I need to read the docs on what is the intent -- I thought that swift 6 had some notion of being able to tell that things were not used concurrently.

Reply:
I have been playing around with this a bit and I think we need to take a different approach:  the model (a Module) can't be Sendable but it can be wrapped in a Sendable type that guards access.  We need the async evaluation and the innards are definitely not thread safe.